cli/wallet/multisig.go

```diff
@@ -11,3 +21,87 @@ type TxWithSignatures struct {
 	// Sigs is a map from a public key to it's corresponding signature.
 	Sigs map[string]string `json:"signatures"`
 }
+
+func newMultisigCommands() []cli.Command {
+	return []cli.Command{
+		{
+			Name:      "sign",
+			Usage:     "sign a transaction",
+			UsageText: "multisig sign --path <path> --in <file.in> --out <file.out>",
+			Action:    signMultisig,
+			Flags: []cli.Flag{
+				walletPathFlag,
+				rpcFlag,
+				timeoutFlag,
+				outFlag,
+				inFlag,
+			},
+		},
+	}
+}
+
+func signMultisig(ctx *cli.Context) error {
+	wall, err := openWallet(ctx.String("path"))
+	if err != nil {
+		return cli.NewExitError(err, 1)
+	}
+	defer wall.Close()
+
+	data, err := ioutil.ReadFile(ctx.String("in"))
+	if err != nil {
+		return cli.NewExitError(fmt.Errorf("can't read input file: %v", err), 1)
+	}
+
+	txSig := new(TxWithSignatures)
+	if err := json.Unmarshal(data, txSig); err != nil {
+		return cli.NewExitError(fmt.Errorf("can't parse transaction: %v", err), 1)
+	}
+	script, err := hex.DecodeString(txSig.Script)
+	if err != nil {
+		return cli.NewExitError(fmt.Errorf("invalid script: %v", err), 1)
+	}
+	shash := hash.Hash160(script)
+	addr := address.Uint160ToString(shash)
+	acc := wall.GetAccount(shash)
+	if acc == nil {
+		return cli.NewExitError(fmt.Errorf("can't find account for the address: %s", addr), 1)
+	}
+
+	tx := &txSig.Tx
+	tx.Data = new(transaction.ContractTX)
```

Comment:
I thought we already have it in `txSig.Tx`, no?

Reply:
`Data` field of `Transaction` isn't marshaled in JSON.

Reply:
Which reminds me of #586. Though `ContractTX` is a special one, nothing bad happens if it's missing.